src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts

```diff
@@ -50,9 +50,34 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 			this._logService.trace('RunAutomaticTasks: Awaiting task system info.');
 			await Event.toPromise(Event.once(this._taskService.onDidChangeTaskSystemInfo));
 		}
-		const workspaceTasks = await this._taskService.getWorkspaceTasks(TaskRunSource.FolderOpen);
+		let workspaceTasks = await this._taskService.getWorkspaceTasks(TaskRunSource.FolderOpen);
 		this._logService.trace(`RunAutomaticTasks: Found ${workspaceTasks.size} automatic tasks`);
-		await this._runWithPermission(this._taskService, this._configurationService, workspaceTasks);
+
+		let autoTasks = this._findAutoTasks(this._taskService, workspaceTasks);
+		this._logService.trace(`RunAutomaticTasks: taskNames=${JSON.stringify(autoTasks.taskNames)}`);
+
+		// As seen in some cases with the Remote SSH extension, the tasks configuration is loaded after we have come
+		// to this point. Let's give it some extra time.
+		if (autoTasks.taskNames.length === 0) {
+			const updatedWithinTimeout = await Promise.race([
+				new Promise<boolean>((resolve) => {
+					Event.toPromise(Event.once(this._taskService.onDidChangeTaskConfig)).then(() => resolve(true));
+				}),
+				new Promise<boolean>((resolve) => {
+					const timer = setTimeout(() => { clearTimeout(timer); resolve(false); }, 10000);
```

Comment:
Waiting on some timeout is suspicious to me. Seems like you should just be able to know for sure when the task configuration is loaded. But I reloaded the window a few times and it worked when I tried it.

Reply:
Many thanks for your review!

I fully agree that this isn't the prettiest solution. In my opinion, it would have been better if TaskService waited with firing the existing onDidChangeTaskSystemInfo event until it had tried to fetch the task configuration, if any. I suspect this is where the hidden timing issue really is. But I know too little about the interactions involved in this code base, also together with the remote SSH extension, to look into that right now. And debugging all this was really quite complex when I couldn't use the remote SSH extension with my custom Code OSS build.

The timer is just there to not wait forever if there is no task configuration at all. Removing the timer (and not waiting just once for the new event) could be a way to support running automatic tasks that are configured manually after having opened the folder. But that's new functionality and not something I'd put in a bug fix. 

Anyway, I just tested the latest Insiders build on Windows, which includes this commit. I can no longer reproduce the issues I had.